libraries/thermocouple/Thermocouple.h

```diff
@@ -97,71 +100,88 @@ namespace Thermocouple {
     }
   }
 
-  // namespace Analog { //TMP36
-  //   // calibration = 25 C at 750mV
-  //
-  //   int DTRDY_PIN = 28;
-  //   int RESET_PIN = 29;
-  //
-  //   // first 7 bits for Arduino Wire
-  //   uint8_t ADC2_ADDR = 0b1001000;
-  //
-  //   float tempOverVoltageScale = 1 / 0.01;
-  //   float voltageOffset = 0.75, tempOffset = 25; // 25 C = 0.450 V
-  //
-  //   long rawRead;
-  //   float voltageRead;
-  //   float tempRead;
-  //
-  //   ADS1219 ads(DTRDY_PIN, ADC2_ADDR, NULL);
-  //
-  //   void init() {
-  //     // ads.begin();
-  //
-  //     ads.setConversionMode(CONTINUOUS);
-  //
-  //     ads.setVoltageReference(REF_EXTERNAL);
-  //
-  //     ads.setGain(GAIN_ONE);
-  //
-  //     ads.setDataRate(90);
-  //   }
-  // 
-  //   void readTemperatureData(float *data) {
-  //     rawRead = ads.readData(2); // thermocouple on AIN1 of ADC2
-  //     // Serial.println(rawRead);
-  //     // Serial.println(ads.readData(2));
-  //     // Serial.println(ads.readData(3));
-  //     voltageRead = (float) rawRead * (5.0 / pow(2,23));
-  //     Serial.println(voltageRead);
-  //     tempRead = ((voltageRead - voltageOffset) * tempOverVoltageScale) + tempOffset;
-  //     data[0] = tempRead;
-  //     data[1] = -1;
-  //   }
-  // }
+  namespace Analog { //TMP36
+
+    // calibration = 25 C at 750mV
+
+    ADS1219 ** _adcs;
+
+    int * _adcIndices; // array of size _numSensors
+    int * _adcChannels;
+
+    int _numSensors; // number of analog thermocouples, not number of adcs
+
+    float tempOverVoltageScale = 1 / 0.01;
+    float voltageOffset = 0.75, tempOffset = 25; // 25 C = 0.450 V
+
+    long rawRead;
+    float voltageRead;
+
+    void init (int numSensors, int * adcIndices, int * adcChannels, ADS1219 ** adcs) {
+      _numSensors = numSensors;
+      _adcIndices = adcIndices;
+      _adcChannels = adcChannels;
+      _adcs = _adcs;
+    }
+
+    // float *data is only of size 7 rn, ensure that we only expect <= 7 readings.
+    void readTemperatureData(float *data) {
+      int i = 0;
+      while (i < _numSensors) {
+        rawRead = _adcs[_adcIndices[i]]->readData(_adcChannels[i]);
+        voltageRead = (float) rawRead * (5.0 / pow(2,23));
+        data[i] = ((voltageRead - voltageOffset) * tempOverVoltageScale) + tempOffset;
+        i++;
+      }
+      data[i] = -1;
```

Comment:
read function is linear in number of sensors of this type instead of linear in total number of Adcs.

Reply:
Changes look great! All around looks pretty clean & understandable, both in config file & in library too. I'll test this build out on the Flight Computer that I have